test/integration/squash_filter_integration_test.cc

```diff
@@ -0,0 +1,309 @@
+#include "test/integration/http_integration.h"
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+
+#define ENV_VAR_VALUE "somerandomevalue"
+
+using Envoy::Protobuf::util::MessageDifferencer;
+
+namespace Envoy {
+
+class SquashFilterIntegrationTestBase : public HttpIntegrationTest,
+                                        public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  using HttpIntegrationTest::HttpIntegrationTest;
+
+  FakeStreamPtr sendSquash(FakeHttpConnectionPtr& fake_squash_connection, std::string status,
+                           std::string body) {
+
+    FakeStreamPtr request_stream = fake_squash_connection->waitForNewStream(*dispatcher_);
+    request_stream->waitForEndStream(*dispatcher_);
+    if (body.empty()) {
+      request_stream->encodeHeaders(Http::TestHeaderMapImpl{{":status", status}}, true);
+    } else {
+      request_stream->encodeHeaders(Http::TestHeaderMapImpl{{":status", status}}, false);
+      Buffer::OwnedImpl responseBuffer(body);
+      request_stream->encodeData(responseBuffer, true);
+    }
+    return request_stream;
+  }
+
+  FakeStreamPtr sendSquashCreate(FakeHttpConnectionPtr& fake_squash_connection, std::string body) {
+    return sendSquash(fake_squash_connection, "201", body);
+  }
+
+  FakeStreamPtr sendSquashOk(FakeHttpConnectionPtr& fake_squash_connection, std::string body) {
+    return sendSquash(fake_squash_connection, "200", body);
+  }
+
+  IntegrationStreamDecoderPtr sendDebugRequest(IntegrationCodecClientPtr& codec_client) {
+    IntegrationStreamDecoderPtr response(new IntegrationStreamDecoder(*dispatcher_));
+    Http::TestHeaderMapImpl headers{{":method", "GET"},
+                                    {":authority", "www.solo.io"},
+                                    {"x-squash-debug", "true"},
+                                    {":path", "/getsomething"}};
+    codec_client->makeHeaderOnlyRequest(headers, *response);
+    return response;
+  }
+
+  /**
+   * Destructor for an individual integration test.
+   */
+  void TearDown() override {
+    test_server_.reset();
+    fake_upstreams_.clear();
+  }
+};
+
+class SquashFilterIntegrationTestV1 : public SquashFilterIntegrationTestBase {
+public:
+  SquashFilterIntegrationTestV1()
+      : SquashFilterIntegrationTestBase(Http::CodecClient::Type::HTTP1, GetParam()) {}
+
+  void SetUp() override {
+    fake_upstreams_.emplace_back(
+        new AutonomousUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
+    registerPort("upstream", fake_upstreams_[0]->localAddress()->ip()->port());
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
+    registerPort("upstream_squash", fake_upstreams_[1]->localAddress()->ip()->port());
+    fake_upstreams_.back()->set_allow_unexpected_disconnects(true);
+
+    ::setenv("SQUASH_ENV_TEST", ENV_VAR_VALUE, 1);
+
+    createTestServer("test/config/integration/squash_filter.json", {"http"});
```

Comment:
We're actively trying to deprecate the v1 config.  We shouldn't extend v1 config or add v1 integration tests unless we have a compelling reason.

Reply:
I wanted to ask Matt and Harvey were we stood on this topic. Thanks for the comment

Reply:
The main reason for v1 config is that this filter is to mainly be used in service mesh scenarios, and itsio doesn't currently support v2. hence the v1 api support.

Reply:
Hm, I'll leave final call up to @mattklein123 but given we've deleted all the v1 filter tests I'd be inclined to leave in the v1 config changes but remove the tests.

If we leave the tests as well I will definitely ask that you combine the test classes and make set up based on a test param, to make it clear that the same tests are running for v1 and v2 (plus avoiding a bunch of duplicate code)

Reply:
I would probably remove the v1 integration test, but please make sure you have v1 conversion tests (just for config load). That doesn't necessarily require a full integration test.